code/modules/procedural mapping/planetGenerator/story_generator/_story_generator.dm

```diff
@@ -0,0 +1,658 @@
+// STORY THEMES
+// ============================================================================
+
+var/list/datum/story_theme/story_themes = list()
+
+/proc/initialize_story_themes()
+	if(story_themes.len)
+		return
+	for(var/theme_type in subtypesof(/datum/story_theme))
+		var/datum/story_theme/ST = new theme_type()
+		story_themes += ST
+
+/proc/get_compatible_story_theme(var/ruin_theme_flags)
+	initialize_story_themes()
+	var/list/compatible = list()
+	for(var/datum/story_theme/ST in story_themes)
+		if(ruin_theme_flags & ST.theme_flag)
+			compatible += ST
+	if(!compatible.len)
+		return null
+	return pick(compatible)
+
+/proc/get_ruins_for_theme(var/theme_flag)
+	var/list/compatible = list()
+	for(var/ruin_type in subtypesof(/datum/map_element/ruin/story))
+		var/datum/map_element/ruin/story/R = ruin_type
+		if(initial(R.theme) & theme_flag)
+			compatible += ruin_type
+	return compatible
+
+// ============================================================================
+// OLD DATABASE
+// ============================================================================
+
+/obj/machinery/old_database
+	icon = 'icons/obj/machines/telecomms.dmi'
+	icon_state = "blackbox_off"
+	name = "Old Data Storage Unit"
+	desc = "An ancient data storage unit from a forgotten era. It looks like it could still be operational with some effort."
+	density = 1
+	anchored = 1.0
+	use_power = MACHINE_POWER_USE_NONE
+	var/activated = FALSE
+	var/activating = FALSE
+	var/datum/story_theme/assigned_theme
+	var/story_year = 0
+	var/character_name = ""
+	var/ruin_name = ""
+
+/obj/machinery/old_database/attack_hand(mob/user)
+	if(activated)
+		to_chat(user, "<span class='notice'>\The [src] has already been activated and its data retrieved.</span>")
+		return
+
+	if(activating)
+		to_chat(user, "<span class='warning'>\The [src] is already in the process of rebooting!</span>")
+		return
+
+	activating = TRUE
+	var/reboot_time = rand(5, 15) MINUTES
+
+	visible_message("<span class='notice'>\The [src] begins to hum as [user] initiates the boot sequence...</span>")
+	playsound(src, 'sound/machines/click.ogg', 50, 1)
+
+	spawn(20)
+		say("REBOOT SEQUENCE INITIATED. ESTIMATED TIME TO FULL SYSTEM RESTORATION: [round(reboot_time/10/60)] MINUTES.")
+		say("PLEASE STAND BY...")
+
+	spawn(reboot_time)
+		complete_activation()
+
+/obj/machinery/old_database/proc/complete_activation()
+	if(activated)
+		return
+
+	activated = TRUE
+	activating = FALSE
+	icon_state = "blackbox"
+
+	var/turf/T = get_turf(src)
+	var/datum/allocation/alloc = SSmapping.get_allocation(trf = T)
+
+	if(istype(alloc))
+		if(alloc.comms_relay.activate())
+			say("PLANETARY RELAY LINK ESTABLISHED.")
+
+	visible_message("<span class='notice'>\The [src] completes its boot sequence with a triumphant chime!</span>")
+	playsound(src, 'sound/machines/ping.ogg', 50, 1)
+	say("SYSTEM RESTORATION COMPLETE. GENERATING DATA ARCHIVE...")
+
+	spawn(30)
+		generate_data_disk()
```

Comment:
Almost useless, like that one syndicate listening satellite. By the time this gets found AND restored, research will be already done (or even maxed).
Works as fluff, but the reward is extremely underwhelming for the time it takes to get.
Could instead generate an "Expedition Research" disk, and gate a couple devices behind this new research.
Alternatively, just make it spawn a disk with xenoarch research.

Reply:
Agreed; this is something I can do down the line but for now it'll just be some token research.